src/lib/bootstrap_controller/bootstrap_controller.ml

```diff
@@ -180,14 +184,17 @@ let external_transition_compare consensus_constants =
   Comparable.lift
     (fun existing candidate ->
       (* To prevent the logger to spam a lot of messsages, the logger input is set to null *)
-      if Consensus.Data.Consensus_state.Value.equal existing candidate then 0
+      if
+        Consensus.Data.Consensus_state.Value.equal (With_hash.data existing)
```

Comment:
Do we want to check the protocol state hashes here instead? It seems like 2 blocks from the same producer but with different resulting ledgers would hit this condition in our attack scenario.

Reply:
Not sure I follow what you mean here. Can you outline the scenario you are envisioning?

This comparison function is only used to order transitions we collected during bootstrap before submitting them to catchup, afaict.

Reply:
A block producer could create e.g. 2 blocks containing different transactions but which are otherwise equal, and it seems that this comparison would treat them as equal (since it only looks at consensus state).

It's probably fine to leave this as-is if this can't happen here.

Reply:
It doesn't matter for this comparison function since we only use it for ordering how we insert them into the frontier. As far as I know, the ordering is only an optimization, not a necessity. But I will change it, just in case someone ends up using this comparison code later for another purpose.